consensus/src/sync/history/cluster.rs

```diff
@@ -366,9 +366,15 @@ impl<TNetwork: Network + 'static> SyncCluster<TNetwork> {
         let current_epoch_number = blockchain.epoch_number();
         let num_known_txs = if epoch_number == current_epoch_number {
             let last_macro_block = Policy::last_macro_block(current_block_number);
-            blockchain
-                .history_store
-                .num_epoch_transactions_before(last_macro_block, None) as u64
+            // Check if Policy::last_macro_block crosses an epoch boundary. This could happen if locally the node is in the first batch of an epoch.
+            // In this case the number of transactions it knows must be 0.
+            if Policy::epoch_at(last_macro_block) != current_epoch_number {
```

Comment:
I think you could use `Policy::is_election_block(last_macro_block)` here, too. It tests for the same.